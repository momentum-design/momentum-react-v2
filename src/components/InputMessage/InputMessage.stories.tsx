import React from 'react';
import { Story } from '@storybook/react';

import InputMessage, { InputMessageProps, INPUT_MESSAGE_CONSTANTS } from './';

import ThemeProvider from '../ThemeProvider';

export default {
  title: 'Momentum UI/InputMessage',
  component: InputMessage,
  argTypes: {
    className: {
      defaultValue: undefined,
      description:
        'If present, the class name will be added to the underlying component. Used to override styles by consumers.',
      control: { type: 'text' },
      table: {
        type: {
          summary: 'string',
        },
        defaultValue: {
          summary: undefined,
        },
      },
    },
    level: {
      description: 'The level of the message.',
      options: INPUT_MESSAGE_CONSTANTS.MESSAGE_LEVEL,
      control: { type: 'select' },
    },
  },
};

const Template: Story<InputMessageProps> = (args) => (
  <ThemeProvider>
    <InputMessage {...args} />
  </ThemeProvider>
);

const Example = Template;

Example.args = {
  message: 'This is a message',
  level: 'error',
};

export { Example };
