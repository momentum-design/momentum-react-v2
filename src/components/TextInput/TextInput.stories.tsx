import React, { useState, FC, useRef } from 'react';
import { MultiTemplate, Template } from '../../storybook/helper.stories.templates';
import ButtonPill from '../ButtonPill';

import TextInput, { TextInputProps } from './';

const messageArrOptions = {
  empty: [],
  error: [{ message: 'Error message', level: 'error' }],
  success: [{ message: 'Success message', level: 'success' }],
  warning: [{ message: 'Warning message', level: 'warning' }],
};

export default {
  title: 'Momentum UI/TextInput',
  component: TextInput,
  args: {
    label: 'Label',
    description: 'Description text for this input',
    initialValue: 'some input',
    placeholder: 'Input placeholder',
  },
  argTypes: {
    messageArr: {
      description: 'The list of messages to be passed in',
      options: Object.keys(messageArrOptions),
      mapping: messageArrOptions,
      control: { type: 'select' },
    },
    description: {
      description: 'The description text for the input',
      control: { type: 'text' },
    },
    isDisabled: {
      description: 'Whether the input is disabled or not',
      control: { type: 'boolean' },
    },
    label: {
      description: 'The label for the input',
      control: { type: 'text' },
    },
    inputMaxLen: {
      description: 'The max amount of characters that this input should allow',
      control: { type: 'number' },
    },
    className: {
      description:
        'If present, the class name will be added to the underlying component. Used to override styles by consumers.',
      control: { type: 'text' },
    },
    clearAriaLabel: {
      description: 'Aria label for the clear button',
      control: { type: 'text' },
    },
    inputClassName: {
      description:
        'If present, the class name will be added to the underlying input. Used to override styles by consumers.',
      control: { type: 'text' },
    },
  },
};

interface StoryProps extends TextInputProps {
  initialValue: string;
}

interface FormProps extends TextInputProps {
  displayName?: boolean;
  emailAddress?: boolean;
  password?: boolean;
}

const PaddedExample: FC<TextInputProps> = (props: TextInputProps) => {
  return (
    <div style={{ margin: '1rem' }}>
      <TextInput style={{ width: '20rem' }} {...props} />
    </div>
  );
};

const BetterExample: FC<StoryProps> = (props: StoryProps) => {
  const mutatedProps = { ...props };
  delete mutatedProps.initialValue;

  const [value, setValue] = useState(props.initialValue);

  const ref = useRef<HTMLInputElement>();
  const onChange = (e) => {
    setValue(e);
    if (ref.current) {
      // eslint-disable-next-line no-console
      console.log(ref.current.value);
    }
  };

  return <TextInput {...mutatedProps} value={value} ref={ref} onChange={onChange} />;
};

const FormExample: FC<FormProps> = (props: FormProps) => {
  const { displayName = false, emailAddress = false, password = false } = props;

  return (
    <form
      onSubmit={(e) => {
        e.preventDefault();
        location.reload();
      }}
    >
      {displayName && (
        <BetterExample
          initialValue=""
          placeholder={'Display Name'}
          type="name"
          style={{ marginTop: '1rem', marginBottom: '1rem', width: '16.25rem' }}
        />
      )}
      {emailAddress && (
        <BetterExample
          initialValue=""
          placeholder={'Email address'}
          type="email"
          style={{
            marginTop: !displayName ? '1rem' : '0',
            marginBottom: '1rem',
            width: '16.25rem',
          }}
        />
      )}
      {password && (
        <BetterExample
          initialValue=""
          type="password"
          placeholder={'Password'}
          style={{ marginBottom: '1rem', width: '16.25rem' }}
        />
      )}
      <ButtonPill type="submit" size={32} grown style={{ marginBottom: '1rem', width: '16.25rem' }}>
        Submit
      </ButtonPill>
    </form>
  );
};

const Example = Template<StoryProps>(BetterExample).bind({});

Example.args = {};

const Forms = Template<FormProps>(() => {
  return (
    <div>
      These stories allow for testing of the behaviour and styling of autofill in browers. The first
      form is to test the autofill of contact information, such as a Display Name.
      <FormExample displayName={true} emailAddress={true} />
      The second form is to test the autofill of log-on credentials and, when available in browser,
      a password suggestion.
      <FormExample emailAddress={true} password={true} />
    </div>
  );
});

Forms.args = {};

const Common = MultiTemplate<TextInputProps>(PaddedExample).bind({});
Common.args = {};
Common.parameters = {
  variants: [
    {
      value: 'test',
      label: 'Label',
      description: 'This is help text for the input.',
    },
    {
      label: 'Label',
      value: 'test',
      isDisabled: true,
      description: 'This is help text for the input.',
    },
    {
      label: 'Label',
      value: 'test',
      description: 'This is help text for the input.',
      messageArr: [{ level: 'error', message: 'Error message' }],
    },
  ],
};

export { Example, Forms, Common };
