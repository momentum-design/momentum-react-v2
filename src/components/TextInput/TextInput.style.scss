.md-text-input-wrapper {
  position: relative;
  text-align: left;

  label {
    margin-bottom: 0.5rem;
    margin-left: 0.75rem;
    display: flex;
  }

  .md-text-input-container {
    background-color: var(--mds-color-theme-background-solid-primary-normal);
    border: 0.0625rem var(--md-globals-border-style-solid) var(--mds-color-theme-outline-input-normal);
    border-radius: 0.5rem;

    &:hover {
      background-color: var(--mds-color-theme-background-primary-hover);
      border-color: var(--mds-color-theme-outline-input-normal);

      input {
        color: var(--mds-color-theme-text-primary-normal);
      }
    }
  }

  input {
    width: calc(100% - 2rem);
    color: var(--mds-color-theme-text-primary-normal);
    outline: none;
    height: 2rem;
    box-sizing: border-box;
    box-shadow: none;
    background-color: transparent;
    border: 0;
    font-family: var(--md-globals-font-default);
    font-size: 1rem;
    padding-left: 0.75rem;

    &::placeholder {
      color: var(--mds-color-theme-text-secondary-normal);
      opacity: 1;
    }

    &::-webkit-contacts-auto-fill-button {
      background-color: var(--mds-color-theme-text-secondary-normal);

      &:hover {
        background-color: var(--mds-color-theme-text-primary-normal);
      }

      &:active {
        background-color: var(--mds-color-theme-text-secondary-normal);
      }
    }

    &:-webkit-autofill,
    &:-webkit-autofill:hover,
    &:-webkit-autofill:focus,
    &:-webkit-autofill:active {
      -webkit-animation: autofill 0s forwards;
      animation: autofill 0s forwards;
      border-radius: 0.438rem 0 0 0.438rem;
    }

    @keyframes autofill {
      100% {
        background: transparent;
        color: inherit;
      }
    }

    @-webkit-keyframes autofill {
      100% {
        background: transparent;
        color: inherit;
      }
    }

    &:-webkit-autofill-strong-password {
      margin-top: 0.0625rem;
      margin-left: 0.0625rem;
      -webkit-box-shadow: 0 0 0 3.75rem #f9ffbd inset;
      box-shadow: 0 0 0 3.75rem #f9ffbd inset;
      border-radius: 0.375rem 0 0 0.375rem;
    }
  }

  &[data-level='error'] {
    .md-text-input-container {
      background-color: var(--mds-color-theme-background-alert-error-normal);
      border-color: var(--mds-color-theme-outline-cancel-normal);

      &:hover {
        background-color: var(--mds-color-theme-background-primary-hover);
        border-color: var(--mds-color-theme-outline-cancel-normal);
      }
    }
  }

  &[data-focus='true'] {
    .md-text-input-container {
      box-shadow: var(--md-globals-focus-ring-box-shadow);
    }
  }

  &[data-disabled='true'] {
    .md-text-input-container {
      background-color: var(--mds-color-theme-background-primary-disabled);
      border-color: var(--mds-color-theme-background-primary-disabled);

      input {
        color: var(--mds-color-theme-text-primary-disabled);
      }
    }
  }

  .md-text-input-help,
  .md-text-input-error {
    margin-left: 0.75rem;
    margin-top: 0.5rem;
  }

  .clear-icon {
    border: 0;
    background-color: transparent;
    cursor: pointer;
    position: absolute;
    color: var(--mds-color-theme-text-secondary-normal);
    display: inline-flex;
    height: 2rem;
    align-items: center;
    outline: none;
  }
}
