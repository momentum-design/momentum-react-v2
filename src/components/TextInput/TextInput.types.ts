import type { AriaTextFieldProps } from '@react-types/textfield';
import { CSSProperties } from 'react';
import type { Message } from '../InputMessage/InputMessage.types';

export interface Props extends Omit<AriaTextFieldProps, 'errorMessage'> {
  /**
   * Optional css class name
   */
  className?: string;
  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;
  /**
   * Optional aria label on the clear button
   */
  clearAriaLabel?: string;
  /**
   * Input css class name string
   */
  inputClassName?: string;
  /**
   * Description text to be displayed below the input
   */
  description?: string;
  /**
   * Optional Input ref prop type
   */
  inputRef?: React.MutableRefObject<HTMLInputElement>;
  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;
  /**
   * Array of Messages with message and type to display below the input
   */
  messageArr?: Message[];
  /**
   * Max amount of characters the input should allow
   */
  inputMaxLen?: number;
}
