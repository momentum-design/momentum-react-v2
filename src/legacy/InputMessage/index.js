/** @component input */

import React from 'react';
import PropTypes from 'prop-types';

/** Input Message with required message */
/**
 * @deprecated - Components in the legacy folder (/src/legacy) are deprecated. Please use a component from the components folder (/src/components) instead. Legacy components may not follow accessibility standards.
 **/
const InputMessage = ({ message }) => {
  return (
    <div className="md-input__message" role="alert">
      {message}
    </div>
  );
};

InputMessage.propTypes = {
  /** @prop message message for InputMessage component */
  message: PropTypes.string.isRequired,
};

InputMessage.displayName = 'InputMessage';

export default InputMessage;
