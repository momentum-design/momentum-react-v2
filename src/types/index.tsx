import type { Button } from '@momentum-design/components/dist/react';

export type MdcButtonProps = React.ComponentProps<typeof Button>;
