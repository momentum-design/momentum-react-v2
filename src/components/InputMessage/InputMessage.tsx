import React, { ReactElement } from 'react';

import './InputMessage.style.scss';
import { STYLE } from './InputMessage.constants';
import { Props, MessageLevel, Message } from './InputMessage.types';

import Icon from '../Icon';
import Text from '../Text';
import classnames from 'classnames';

const messagePriority = {
  none: 0,
  success: 1,
  warning: 2,
  error: 3,
};

const determineMessageLevel = (array: Message[]): MessageLevel => {
  return array.reduce((agg, e: Message) => {
    return messagePriority[agg] > messagePriority[e.level] ? agg : e.level || 'none';
  }, 'none');
};

const filterMessagesByLevel = (array, value) => {
  return array.reduce((agg, e) => (e.level === value ? agg.concat(e.message) : agg), []);
};

type GetFilteredMessagesReturn = [MessageLevel, string[]];

export const getFilteredMessages = (allMessages: Message[]): GetFilteredMessagesReturn => {
  const messageLevel: MessageLevel =
    (allMessages.length > 0 && determineMessageLevel(allMessages)) || 'none';
  const messages = (messageLevel && filterMessagesByLevel(allMessages, messageLevel)) || null;
  return [messageLevel, messages];
};

const InputMessage = (props: Props): ReactElement => {
  const { message, level = 'none', className, id } = props;

  return (
    <div className={classnames(STYLE.wrapper, className)} id={id}>
      <div className={STYLE.message} message-level={level}>
        {(level == 'error' || level == 'warning') && (
          <div className={STYLE.icon}>
            <Icon
              name="warning"
              weight="bold"
              scale={16}
              fillColor={`var(--mds-color-theme-text-${level}-normal)`}
            />
          </div>
        )}
        <Text type="body-secondary" tagName="small" className={STYLE.text}>
          {message}
        </Text>
      </div>
    </div>
  );
};

/**
 * Message for use under Inputs that shows text with icon and color corresponding to level
 */

export default InputMessage;
