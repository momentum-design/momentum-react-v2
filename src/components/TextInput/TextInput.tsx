/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { ReactElement, InputHTMLAttributes, RefObject, forwardRef, useRef } from 'react';
import { useTextField } from '@react-aria/textfield';
import { useSearchFieldState } from '@react-stately/searchfield';
import classnames from 'classnames';

import './TextInput.style.scss';
import { Props } from './TextInput.types';
import InputMessage, { getFilteredMessages } from '../InputMessage';
import { ButtonSimple, Icon } from '..';
import { STYLE } from './TextInput.constants';
import { useFocusState } from '../../hooks/useFocusState';
import { v4 as uuidV4 } from 'uuid';

const TextInput = (props: Props, ref: RefObject<HTMLInputElement>): ReactElement => {
  const {
    messageArr = [],
    label,
    className,
    clearAriaLabel,
    description,
    inputClassName,
    isDisabled,
    style,
    id,
    inputMaxLen,
  } = props;

  const componentRef = React.useRef<HTMLInputElement>();
  const inputRef = ref || componentRef;

  const [messageType, messages] = getFilteredMessages(messageArr);
  const errorMessage = messageType === 'error' ? messages[0] : undefined;

  const { labelProps, inputProps, descriptionProps, errorMessageProps } = useTextField(
    { ...props, errorMessage },
    inputRef
  );

  const { isFocused, focusProps } = useFocusState(props);
  const state = useSearchFieldState(props);

  const messageId = useRef(uuidV4());
  const labelId = useRef(uuidV4());
  const clearButtonId = useRef(uuidV4());

  const onClearButtonPress = () => {
    state.setValue('');
    if (inputRef.current) {
      inputRef.current.focus();
    }
  };

  const handleClick = () => {
    if (inputRef.current) {
      inputRef.current.focus();
    }
  };

  return (
    <div
      data-level={messageType}
      data-focus={isFocused}
      data-disabled={isDisabled}
      style={style}
      onClick={handleClick}
      className={classnames(STYLE.wrapper, className)}
    >
      {label && (
        <label {...labelProps} htmlFor={labelProps.htmlFor} id={labelId.current}>
          {label}
        </label>
      )}
      <div className={STYLE.container}>
        <input
          {...(inputProps as InputHTMLAttributes<HTMLInputElement>)}
          {...focusProps}
          className={inputClassName}
          ref={inputRef}
          maxLength={inputMaxLen}
          aria-describedby={messages && !!messages.length ? messageId.current : undefined}
        />
        {!!state.value && !isDisabled && (
          <ButtonSimple
            className="clear-icon"
            aria-label={clearAriaLabel}
            onPress={onClearButtonPress}
            id={clearButtonId.current}
            // If there is a visible label, then sr should read for this x button "clearAriaLabel (Ex. 'Clear input') + visible label (Ex. 'Password')"
            // Else, if input was given an id and ideally has an aria-label, then sr should read for this x button "clearAriaLabel (Ex. 'Clear input') + input aria-label (Ex. 'Password')"
            // Else, sr should read for this x button "clearAriaLabel (Ex. 'Clear input')"
            aria-labelledby={
              label
                ? `${clearButtonId.current} ${labelId.current}`
                : id
                ? `${clearButtonId.current} ${id}`
                : clearButtonId.current
            }
          >
            <Icon scale={18} name="cancel" />
          </ButtonSimple>
        )}
      </div>
      {!!description && !messages?.length && (
        <InputMessage
          className={STYLE.help}
          level="help"
          {...descriptionProps}
          message={description}
        />
      )}
      {messages && !!messages.length && (
        <div {...errorMessageProps}>
          {messages.map((m, i) => (
            <InputMessage
              className={STYLE.error}
              message={m}
              key={`input-message-${i}`}
              level={messageType}
              id={messageId.current}
            />
          ))}
        </div>
      )}
    </div>
  );
};

/**
 * Short text input
 */

const _TextInput = forwardRef(TextInput);

_TextInput.displayName = 'TextInput';

export default _TextInput;
