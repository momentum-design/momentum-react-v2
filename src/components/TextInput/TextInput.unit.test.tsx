import React from 'react';
import { SSRProvider } from '@react-aria/ssr';

import { mountAndWait } from '../../../test/utils';
import TextInput, { TEXT_INPUT_CONSTANTS as CONSTANTS } from './';
import { act } from 'react-dom/test-utils';
import { Message } from '../InputMessage/InputMessage.types';
import InputMessage from '../InputMessage';

jest.mock('uuid', () => {
  return {
    v4: () => 'desc-test-ID',
  };
});

describe('<TextInput/>', () => {
  describe('snapshot', () => {
    const mountComponent = async (component) => {
      const container = await mountAndWait(<SSRProvider>{component}</SSRProvider>);
      return container;
    };

    it('should match snapshot', async () => {
      expect.assertions(1);

      const container = await mountComponent(<TextInput aria-label="text-input" />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = await mountComponent(
        <TextInput aria-label="text-input" className={className} />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = await mountComponent(<TextInput aria-label="text-input" id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with label', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = await mountComponent(<TextInput label="Password" id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with clearAriaLabel', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = await mountComponent(
        <TextInput label="Password" clearAriaLabel="Clear this input" id={id} />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = await mountComponent(<TextInput aria-label="text-input" style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with error text', async () => {
      expect.assertions(1);

      const message: Message = { message: 'test', level: 'error' };

      const container = await mountComponent(
        <TextInput aria-label="text-input" messageArr={[message]} />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with description', async () => {
      expect.assertions(1);

      const container = await mountComponent(
        <TextInput aria-label="text-input" description="some input" />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with inputMaxLen', async () => {
      expect.assertions(1);

      const container = await mountComponent(<TextInput aria-label="text-input" inputMaxLen={4} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isDisabled', async () => {
      expect.assertions(1);

      const container = await mountComponent(<TextInput aria-label="text-input" isDisabled />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', async () => {
      expect.assertions(1);

      const element = (await mountAndWait(<TextInput aria-label="text-input" />))
        .find(TextInput)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = (
        await mountAndWait(<TextInput aria-label="text-input" className={className} />)
      )
        .find(TextInput)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', async () => {
      expect.assertions(1);

      const id = 'example-id-2';

      const element = (await mountAndWait(<TextInput aria-label="text-input" id={id} />)).find(
        TextInput
      );

      expect(element.props()).toMatchObject({ 'aria-label': 'text-input', id: 'example-id-2' });
    });

    it('should have provided style when style is provided', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = (await mountAndWait(<TextInput aria-label="text-input" style={style} />))
        .find(TextInput)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided inputMaxLen when inputMaxLen is provided', async () => {
      expect.assertions(1);

      const inputMaxLen = 8;
      const inputClassName = 'fake-class-name';

      const element = (
        await mountAndWait(
          <TextInput
            aria-label="text-input"
            inputMaxLen={inputMaxLen}
            inputClassName={inputClassName}
          />
        )
      )
        .find(`.${inputClassName}`)
        .getDOMNode();

      expect(element.getAttribute('maxLength')).toBe(`${inputMaxLen}`);
    });

    it('should have aria-describedby and id when message is provided', async () => {
      expect.assertions(2);

      const textInputComponent = (
        await mountAndWait(<TextInput aria-label="text-input" aria-describedby={'desc-test-ID'} />)
      ).find(TextInput);

      const inputMessageComponent = (
        await mountAndWait(<InputMessage className="error" level="error" id={'desc-test-ID'} />)
      ).find(InputMessage);

      expect(inputMessageComponent.props().id).toStrictEqual('desc-test-ID');
      expect(textInputComponent.props()).toMatchObject({
        'aria-label': 'text-input',
        'aria-describedby': 'desc-test-ID',
      });
    });

    it('should not have aria-labelledby when message is not provided', async () => {
      expect.assertions(1);

      const textInputComponent = (await mountAndWait(<TextInput aria-label="text-input" />)).find(
        TextInput
      );

      expect(textInputComponent.props()['aria-describedby']).toBe(undefined);
    });

    it('should not have the clear input button displayed when value is empty', async () => {
      const textInputComponent = (
        await mountAndWait(<TextInput aria-label="text-input" value="" />)
      ).find(TextInput);

      expect(textInputComponent.find('.clear-icon').exists()).toBe(false);
    });

    it.each(['hello world', '0', '123', ' '])(
      'should have the clear input button displayed when not disabled (value = "%s")',
      async (value) => {
        const textInputComponent = (
          await mountAndWait(<TextInput aria-label="text-input" value={value} />)
        ).find(TextInput);

        expect(textInputComponent.find('.clear-icon').exists()).toBe(true);
      }
    );

    it.each(['', 'hello world', '0', '123', ' '])(
      'should not have the clear input button displayed when isDisabled (value = "%s")',
      async (value) => {
        const textInputComponent = (
          await mountAndWait(<TextInput aria-label="text-input" value={value} isDisabled />)
        ).find(TextInput);

        expect(textInputComponent.find('.clear-icon').exists()).toBe(false);
      }
    );
  });

  describe('actions', () => {
    it('clicking on another part of the component gives focus to the input', async () => {
      expect.assertions(1);

      const wrapper = await mountAndWait(<TextInput aria-label="text-input" />);

      const inputElement = wrapper.find('input');
      const inputWrapper = wrapper.find(`.${CONSTANTS.STYLE.wrapper}`);

      const domNode = inputElement.getDOMNode() as HTMLInputElement;
      const focusSpy = jest.spyOn(domNode, 'focus');

      await act(async () => {
        inputWrapper.simulate('click');
      });

      expect(focusSpy).toBeCalledWith();
    });
  });
});
