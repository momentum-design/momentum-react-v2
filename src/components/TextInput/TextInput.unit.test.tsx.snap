// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> snapshot should match snapshot 1`] = `
<SSRProvider>
  <TextInput
    aria-label="text-input"
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
    >
      <div
        className="md-text-input-container"
      >
        <input
          aria-label="text-input"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with className 1`] = `
<SSRProvider>
  <TextInput
    aria-label="text-input"
    className="example-class"
  >
    <div
      className="md-text-input-wrapper example-class"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
    >
      <div
        className="md-text-input-container"
      >
        <input
          aria-label="text-input"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with clearAriaLabel 1`] = `
<SSRProvider>
  <TextInput
    clearAriaLabel="Clear this input"
    id="example-id"
    label="Password"
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
    >
      <label
        htmlFor="test-ID"
        id="desc-test-ID"
      >
        Password
      </label>
      <div
        className="md-text-input-container"
      >
        <input
          aria-labelledby="test-ID"
          disabled={false}
          id="example-id"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with description 1`] = `
<SSRProvider>
  <TextInput
    aria-label="text-input"
    description="some input"
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
    >
      <div
        className="md-text-input-container"
      >
        <input
          aria-label="text-input"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
      <InputMessage
        className="md-text-input-help"
        id={null}
        level="help"
        message="some input"
      >
        <div
          className="md-input-message-wrapper md-text-input-help"
          id={null}
        >
          <div
            className="md-input-message-message"
            message-level="help"
            role="alert"
          >
            <Text
              className="md-input-message--text"
              type="body-secondary"
            >
              <small
                className="md-text-wrapper md-input-message--text"
                data-type="body-secondary"
              >
                some input
              </small>
            </Text>
          </div>
        </div>
      </InputMessage>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with error text 1`] = `
<SSRProvider>
  <TextInput
    aria-label="text-input"
    messageArr={
      [
        {
          "level": "error",
          "message": "test",
        },
      ]
    }
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="error"
      onClick={[Function]}
    >
      <div
        className="md-text-input-container"
      >
        <input
          aria-describedby="desc-test-ID"
          aria-label="text-input"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
      <div
        id={null}
      >
        <InputMessage
          className="md-text-input-error"
          id="desc-test-ID"
          key="input-message-0"
          level="error"
          message="test"
        >
          <div
            className="md-input-message-wrapper md-text-input-error"
            id="desc-test-ID"
          >
            <div
              className="md-input-message-message"
              message-level="error"
              role="alert"
            >
              <div
                className="md-input-message--icon"
              >
                <Icon
                  fillColor="var(--mds-color-theme-text-error-normal)"
                  name="warning"
                  scale={16}
                  weight="bold"
                >
                  <div
                    aria-hidden="true"
                    className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-autoscale={false}
                      data-scale={16}
                      data-test="warning"
                      height="100%"
                      style={
                        {
                          "fill": "var(--mds-color-theme-text-error-normal)",
                        }
                      }
                      viewBox="0, 0, 32, 32"
                      width="100%"
                    />
                  </div>
                </Icon>
              </div>
              <Text
                className="md-input-message--text"
                type="body-secondary"
              >
                <small
                  className="md-text-wrapper md-input-message--text"
                  data-type="body-secondary"
                >
                  test
                </small>
              </Text>
            </div>
          </div>
        </InputMessage>
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with id 1`] = `
<SSRProvider>
  <TextInput
    aria-label="text-input"
    id="example-id"
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
    >
      <div
        className="md-text-input-container"
      >
        <input
          aria-label="text-input"
          disabled={false}
          id="example-id"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with inputMaxLen 1`] = `
<SSRProvider>
  <TextInput
    aria-label="text-input"
    inputMaxLen={4}
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
    >
      <div
        className="md-text-input-container"
      >
        <input
          aria-label="text-input"
          disabled={false}
          id="test-ID"
          maxLength={4}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with label 1`] = `
<SSRProvider>
  <TextInput
    id="example-id"
    label="Password"
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
    >
      <label
        htmlFor="test-ID"
        id="desc-test-ID"
      >
        Password
      </label>
      <div
        className="md-text-input-container"
      >
        <input
          aria-labelledby="test-ID"
          disabled={false}
          id="example-id"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;

exports[`<TextInput/> snapshot should match snapshot with style 1`] = `
<SSRProvider>
  <TextInput
    aria-label="text-input"
    style={
      {
        "color": "pink",
      }
    }
  >
    <div
      className="md-text-input-wrapper"
      data-focus={false}
      data-level="none"
      onClick={[Function]}
      style={
        {
          "color": "pink",
        }
      }
    >
      <div
        className="md-text-input-container"
      >
        <input
          aria-label="text-input"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          type="text"
        />
      </div>
    </div>
  </TextInput>
</SSRProvider>
`;
